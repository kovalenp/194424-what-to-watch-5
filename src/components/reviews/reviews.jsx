import React from "react";
import PropTypes from "prop-types";

import { reviewProps } from "../../validation/propTypes";

const Reviews = (props) => {

  const { reviews } = props;

  const getColumnsData = (arr) => {
    const chunk = Math.round(arr.length / 2);
    return { first: arr.slice(0, chunk), second: arr.slice(chunk) };
  };

  const renderColumn = (data) => (
    <div className="movie-card__reviews-col">
        {data.map((rev, i) => (
          <div className="review" key={rev.date + rev.author + i}>
                  <blockquote className="review__quote">
              <p className="review__text">{rev.text}</p>
                    <footer className="review__details">
                      <cite className="review__author">{rev.author}</cite>
                      <time className="review__date" dateTime="2016-12-24">{rev.date}</time>
                    </footer>
                  </blockquote>

            <div className="review__rating">{rev.rating}</div>
          </div>
        ))}
      </div>
  );

  const { first, second } = getColumnsData(reviews);

  return (
    <div className="movie-card__reviews movie-card__row">
      {first.length > 0 && renderColumn(first)}
      {second.length > 0 && renderColumn(second)}
    </div>
  );

};

Reviews.propTypes = {
  reviews: PropTypes.arrayOf(reviewProps)
};

export default Reviews;
