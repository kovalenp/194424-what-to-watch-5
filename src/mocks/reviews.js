export default [
  {
    "id": 1,
    "user": {"id": 1, "name": `pavel`},
    "rating": 5,
    "comment": `HTMLAcademy B21 criteria is my favorite`,
    "date": `2020-11-15T19:26:30.670Z`
  },
];
