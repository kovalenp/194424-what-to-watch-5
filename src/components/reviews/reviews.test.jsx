import React from "react";
import renderer from "react-test-renderer";
import Reviews from "./reviews";
import reviews from "../../mocks/reviews";

describe(`Reviews meets B21`, () => {

  it(`and rendered correctly for auth'ed`, () => {
    const tree = renderer.create((
      <Reviews
        reviews={reviews} />
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
