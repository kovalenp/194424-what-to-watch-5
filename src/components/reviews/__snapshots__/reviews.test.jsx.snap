// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews meets B21 and rendered correctly for auth'ed 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          HTMLAcademy B21 criteria is my favorite
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            pavel
          </cite>
          <time
            className="review__date"
          >
            November 15, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        5
      </div>
    </div>
  </div>
</div>
`;
